src/dynamics/video_player/player/states.js

```diff
@@ -891,6 +883,31 @@ Scoped.define("module:VideoPlayer.Dynamics.PlayerStates.PlayVideo", [
             this.listenOn(this.dyn, "error:video", function() {
                 this.next("ErrorVideo");
             }, this);
+            this.addManualFallbackListener();
+        },
+
+        getAdPosition: function() {
+            const currentPosition = this.dyn.getCurrentPosition();
+            if (currentPosition === 0) {
+                return State.ADS_POSITIONS.PREROLL
+            } else if (Math.abs(currentPosition - this.dyn.get("duration")) < 0.1) {
+                return State.ADS_POSITIONS.POSTROLL
+            } else {
+                return State.ADS_POSITIONS.MIDROLL
+            }
+        },
+
+        addManualFallbackListener: function() {
+            this.listenOn(this.dyn, "manualFallback", (fallbackUrl) => {
+                this.dyn.set('adtagurl', fallbackUrl);
+                this.dyn.scopes.adsplayer.execute(`requestAds`);
+                this.listenOnce(this.dyn.channel("ads"), "adsManagerLoaded", function() {
+                    this.next("LoadAds", {
+                        position: this.getAdPosition(),
+                        autoplay: true
```

Comment:
Without `autoplay` true, the fallback ad does not play for the preroll on the next playlist item.